pyenvbuilder/commands/check.py

```diff
@@ -1,88 +1,85 @@
 '''
-Module for YAML file validation
+
+Class for YAML file validation
 '''
 from codecs import open
 import errno
 from jsonschema import validate, exceptions
 import yaml
 import logging
 from pathlib import Path
+from .interface import Command
 
 
-logger = logging.getLogger(__name__)
-schema_dir = Path(__file__).absolute().parents[0] / 'schema.yml'
-
-
-def check(**kwargs):
-    '''
-    Validates if proper files or directories were passed in
-    '''
-    # get the 'files' from the kwargs 'check' command
-    get_files = (v for k, v in kwargs.items() if k == 'files')
-    for file_arg in next(get_files):
-        arg_path = Path(file_arg)
-        # support both YAML file formats
-        exts = ['.yml', '.yaml']
+class Check(Command):
+    def __init__(self):
+        self.name = 'check'
+        self.help = 'Validates an YAML file or multiple files'
 
-        if arg_path.exists():
-            # validates one YAML filei
-            if arg_path.is_file() and arg_path.suffix in exts:
-                yaml_validator(arg_path)
-            # validates a directory of YAML files
-            elif arg_path.is_dir():
-                for f in Path(arg_path).rglob('*'):
-                    if f.suffix in exts:
-                        yaml_validator(f)
-        else:
-            logger.error('Invalid path or file name: {}'.format(arg_path))
+        self.logger = logging.getLogger(__name__)
```

Comment:
No need for the logger here if we will use the `__name__`. You can create it above and just call it at your code as a module variable.